op-batcher/batcher/driver.go

```diff
@@ -407,6 +408,45 @@ func (l *BatchSubmitter) mainLoop(ctx context.Context, receiptsCh chan txmgr.TxR
 	}
 }
 
+func (l *BatchSubmitter) waitNodeSyncAndClearState() {
+	// Wait for any in flight transactions
+	// to be ingested by the node before
+	// we start loading blocks again.
+	err := l.waitNodeSync()
+	if err != nil {
+		l.Log.Warn("error waiting for node sync", "err", err)
+	}
+	l.clearState(l.shutdownCtx)
+}
+
+// checkExpectedProgress uses the supplied syncStatus to infer
+// whether the node providing the status has made the expected
+// safe head progress given fully submitted channels held in
+// state.
+func (l *BatchSubmitter) checkExpectedProgress(syncStatus eth.SyncStatus) error {
+
+	// Define a buffer: a delay denominated in L1 blocks
+	// to allow verifier to process blocks.
+	// Set to 0 to require immediate safe head advancement.
+	// Set to 1 to allow e.g. 12s for the safe head to advance
+	// after the L1 inclusion block was ingested.
+	// TODO extract this into a config variable
+	// TODO should we also wait numConfirmations (this is a txmgr config variable, not in scope of the batch submitter)
```

Comment:
What we really want to know is the verifier numConfirmations, but I don't think we can get access to that. We probably have to proceed with our own conservative buffer / confirmation depth. 

When this is a CLI config variable, it can be set in a coordinated way with how the verifier node / sequencer is configured.